rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2217,19 +2238,23 @@ void rskTxWaitingForSignature_fail_adding_an_already_existence_key_after_rskip_3
             .build();
         bridgeSupport.updateCollections(tx);
 
-        // Since only one pegout per updateCollection call is move to waitingForisgnatures
-        // -assert one of the two pegout get moved to pegoutWaitingForSignate and one is left on waitingForConfirmation
-        assertEquals(1, provider.getPegoutsWaitingForConfirmations().getEntries().size());
-        assertEquals(1, provider.getPegoutsWaitingForSignatures().size());
+        /*
+            Since only one pegout per updateCollection call is move to waitingForSignatures
+            assert one of the two pegout get moved to pegoutWaitingForSignate and one is left on waitingForConfirmation
+         */
+        assertEquals(1, pegoutsWaitingForConfirmations.getEntries().size());
+        assertEquals(1, pegoutWaitingForSignatures.size());
+
+        // assert the migration tx was the first one to be confirmed
+        assertTrue(pegoutWaitingForSignatures.containsKey(migrationTx.getPegoutCreationRskTxHash()));
```

Comment:
Not sure this is asserting what we expect it to. Both transactions have the same pegoutCreationRskTxHash so this doesn't really help distinguish which one was confirmed. Maybe we could get the transaction from pegoutWaitingForSignatures and compare it against the migration transaction obtained above

Reply:
Good catch man. You are absolutely right. This assertion is worthless using the `pegoutCreationRskTxHash`, need to get the `btcTx` and compare it against the hash of the migrationTx.